src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,124 @@
+        }
+    }
+
+    /**
+     * extractTask extract a task name from input_parts.
+     * @param input_parts array of strings derived from input
+     * @return task name
+     */
+    private static String extractTask(String[] input_parts) {
+        String task = "";
+        for (int i = 1; i < input_parts.length; i++) {
+            task += input_parts[i] + " ";
+        }
+        return task.trim();
+    }
+
+    /**
+     * Extract details of deadline and event tasks from input.
+     * @param input_parts input in form of String array
+     * @return details of task
+     */
+    private static String extractDetails(String[] input_parts) {
+        String task = "";
+        for (int i = 0; i < input_parts.length; i++) {
+            task += input_parts[i] + " ";
+        }
+        return task.trim();
+    }
+    
+    /**
+     * parseTask creates a Task object based on user input String.
+     * @param input String of user input
+     * @return corresponding Task object
+     */
+    public static Task parseTask(String input) throws Exception { // catch duke exception from extractTask and categorize, throw to op()
+        String[] parts = input.split(" ");
+        TaskType type;
+        try {
+            type = categorize(parts);
+        } catch (UndefinedWordException e) {
+            throw e;
+        }
+
+        if (type == TaskType.ToDo) {
+            parts = input.split(" ");
+            if (parts.length <= 1) {
+                throw new NoDescriptionException("☹ OOPS!!! The description of a todo cannot be empty.\n");
+            } // if not, chillax and continue
+            String name = extractTask(parts);
+            return new ToDo(name);
+        } else if (type == TaskType.Deadline) {
+            if (input.split("/by").length <= 1) {
+                throw new NoDescriptionException("☹ OOPS!!! The description of a deadline cannot be empty.\n");
+            } // if not, chillax and continue
+            String name = extractTask(input.split("/by")[0].split(" "));
+            String deadline = extractDetails(input.split("/by")[1].split(" "));
+            return new Deadline(name, deadline); 
+        } else {
+            if (input.split("/at").length <= 1) {
+                throw new NoDescriptionException("☹ OOPS!!! The description of an event cannot be empty.\n");
+            } // if not, chillax and continue
+            String name = extractTask(input.split("/at")[0].split(" "));
+            String details = extractDetails(input.split("/at")[1].split(" "));
+            
+            return new Event(name, details);
+        }
+    }
+
+    /**
+     * Series of is* methods help identify what operation is specified by the
+     * user.
+     * @param input user input, to be compared with the operation keyword
+     * @return boolean indicating whether it is some operation
+     */
+
+    public static boolean isBye(String input) {
```

Comment:
Maybe just using input.equals() will be clearer since it is only important to check equality and sorting is not required?